src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -50,10 +50,13 @@ public static class Renamer
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var conflictResolution = await locations.ResolveConflictsAsync(
+            var resolution = await locations.ResolveConflictsAsync(
                 newName, nonConflictSymbols, cancellationToken).ConfigureAwait(false);
 
-            return conflictResolution.NewSolution;
+            if (resolution.ErrorMessage != null)
+                throw new ArgumentException(resolution.ErrorMessage);
```

Comment:
>throw new ArgumentException(resolution.ErrorMessage); [](start = 16, length = 53)

Contract

Reply:
This is the behavior of this public api call that I'm preserving. If you try to rename something that you're not allowed to, you get this exception. 

What I changed was not throwing feel inside rename, but instead at the top point. This is necessary for oop as well as we can't throw exceptions across the oop boundary.

Reply:
The same arguments re not throwing ArgumentException from non-public APIs apply as we had discussed previously. In this case I think it would make sense to preserve the error message in the results all the way to the public API entry point and throw there.

Reply:
Doing so in a follow up is fine, btw, as you're preserving behavior.

Reply:
sure.  i'll do that.

Reply:
Done in https://github.com/dotnet/roslyn/pull/43592